pkg/log/zap/flags.go

```diff
@@ -29,16 +29,15 @@ import (
 )
 
 var levelStrings = map[string]zapcore.Level{
-	"debug":  zap.DebugLevel,
-	"-1":     zap.DebugLevel,
-	"info":   zap.InfoLevel,
-	"0":      zap.InfoLevel,
-	"error":  zap.ErrorLevel,
-	"2":      zap.ErrorLevel,
-	"dpanic": zap.DPanicLevel,
-	"panic":  zap.PanicLevel,
-	"warn":   zap.WarnLevel,
-	"fatal":  zap.FatalLevel,
+	"debug": zap.DebugLevel,
+	"info":  zap.InfoLevel,
+	"error": zap.ErrorLevel,
+}
+
+var stackLevelStrings = map[string]zapcore.Level{
+	"disabled": zap.FatalLevel,
```

Comment:
why disabled is `FatalLevel`?

Reply:
I tend to agree that this could be confusing.

@hasbro17 @bharathi-tenneti and I had an offline conversation where we discussed that since `Fatal` is not a valid logging level with `logr`, we didn't want to include it in the set of log level values handled by `--zap-stacktrace-level`. But there's still a desire to be able to disable stacktraces entirely. The best way to do that would be to NOT add `zap.AddStacktrace()` to the set of `ZapOpts` [here](https://github.com/kubernetes-sigs/controller-runtime/blob/add0b6444f5091f56824b070d28f930090d8deab/pkg/log/zap/zap.go#L190).

But given the existing Options fields and defaults, it isn't immediately clear how to do that.

 See https://github.com/kubernetes-sigs/controller-runtime/pull/991#discussion_r447704765

Reply:
We can add a condition to check for `zapcore.FatalLevel` and apply `zap.AddStackTrace()` accordingly. However, its still a question, that we need to provide some level if not for `fatal` to identify `disabled`.

Reply:
Responded in https://github.com/kubernetes-sigs/controller-runtime/pull/991#discussion_r451194763
Let's handle this in a follow up PR.